src/app/CommandSender.h

```diff
@@ -149,14 +246,18 @@ class CommandSender final : public Messaging::ExchangeDelegate
      */
     template <typename CommandDataT>
     CHIP_ERROR AddRequestData(const CommandPathParams & aCommandPath, const CommandDataT & aData,
-                              const Optional<uint16_t> & aTimedInvokeTimeoutMs)
+                              const Optional<uint16_t> & aTimedInvokeTimeoutMs,
+                              AdditionalCommandParameters aOptionalArgs = AdditionalCommandParameters())
```

Comment:
More passing by value here, and various places below.

Reply:
I agree with all the other pass by value, but this one might be a little tricky. In order to do this properly we will need 2, one with the `AdditionalCommandParameters` arg, and one without:
```
template <typename CommandDataT>
CHIP_ERROR AddRequestData(...)
{
   ....
```

Because of the template we are now increasing the memory footprint. I could test out how much of an impact this will have, but it was the reason why I didn't duplicate this method like I did for `PrepareCommand` and `FinishCommand`. Now it might be possible we can remove this template somehow. I was already spending some time thinking about it, but I didn't have any non-trivial ideas, and would rather do this as a follow up sometime maybe in Jan

Reply:
You _should_ be able to pass by const reference and still default it, no?

If we want a non-const ref, then I agree we need two different functions: one takes the arg and one does not and puts an AdditionalCommandParameters on the stack.  I dont think this should particular increase the memory footprint compared to the above; should compile to the same thing.

Reply:
Hm... I never got a size report. Anyways I have a plan to possible remove the templates that I might get around to in Jan time frame.